Adding more explicit error code

yunquan812 opened the pull request:
Summary: {title}, the `Internal` and `InvalidArgument` errors don't tell what is really going wrong

Differential Revision: D78233596

JacobSzwejbka commented on runtime/core/error.h:
```diff
@@ -96,6 +96,12 @@ enum class Error : error_code_t {
+  /*
+   * Registration errors.
+   */
+  RegistrationExceedingMaxKernels = 0x40,
```
They are loosely grouped by type with jumps to allow new types in. If anything the original OperatorMissing is in the wrong spot, but since its there already probably just make this error 0x15 @yunquan812

yunquan812 replied:
Agreed. Let me update it :)

yunquan812 committed:
Adding more explicit error code (#12427)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/12427

{title}, the `Internal` and `InvalidArgument` errors don't tell what is really going wrong

Reviewed By: swolchok

Differential Revision: D78233596
